src/js/clock.js

```diff
@@ -26,9 +21,14 @@ App.Clock.prototype.setTime = function() {
 
     if (currentTime !== this.time) {
         this.time = currentTime;
-
-        document.getElementById('clock').innerText = currentTime;
+        this.clockElement.innerText = currentTime;
     }
-}
+};
+
+App.Clock.prototype.update = function () {
+    this.setTime();
+    requestAnimationFrame(this.update.bind(this));
```

Comment:
Okay, that's a step in the right direction! But in its current state, it has worse performance. That's because the `requestAnimationFrame` is designed to synchronize updates with the browser's repaint cycle, which typically runs at 60 frames per second. This means the update method will be called much more frequently than once per second (as it was designed previously with the `setInterval`). To validate this problem, add a `console.log` in the `setTime` method and see how many times it executes per second (maaaaany).

You need to combine the `requestAnimationFrame` with another approach to reduce these way too many updates.